src/mixer/samplerbank.cpp

```diff
@@ -33,28 +44,40 @@ SamplerBank::SamplerBank(PlayerManager* pPlayerManager)
 SamplerBank::~SamplerBank() {
 }
 
+const QString SamplerBank::samplerFiletype() {
+    return tr("Mixxx Sampler Banks (*.xml)");
+}
```

Comment:
Doesn't feel this is the smartest way to define a `tr` string which is used multple times.
Any other ways to do that?

Reply:
QT_TR_NOOP seems to be the weapon of choice, put in the anonymous namespace.
`const QString kSamplerFileType = QT_TR_NOOP("Mixxx Sampler Banks (*.xml)");`

Let me know what you think.

Reply:
Shouldn't `const QString kSamplerFileType = QObject::tr("Mixxx Sampler Banks (*.xml)"); work just as well? According to Qt docs, `QT_TR_NOOP` marks a string for "delayed translation", but I don't know what that means, maybe lazy loading of the translated string on first use?

Reply:
I'll try to test the translation for both to be sure. Never did that before though..

Reply:
Does nothing: 
```#define QT_TR_NOOP(x) x```
https://github.com/qt/qtbase/blob/eba9196304dcc9129a7bb68db87c6fb42c953090/src/corelib/global/qglobal.h#L1356

But ... lupdate finds it so that later use of tr() with this variable can be translated. 

This is demontarted here: 
https://github.com/qt/qtbase/blob/fb33e2a8e82cfc28cf3c3c4bad657e0400640cb2/src/corelib/doc/snippets/code/src_corelib_global_qglobal.cpp#L389

Reply:
This means the current code translates the string on every call. The proposed solution of @Swiftb0y only translates one time. However I am not sure if the translation is initialized before the string is initialized.

Reply:
Unfortunately I'm not familiar with the translation workflow, so I appreciate any advice how to proceed here.